api/src/libs/usersBusinessesMigration.ts

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BusinessesDataClient, UserDataClient } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { DynamoDBStreamEvent } from "aws-lambda";
+
+export const usersBusinessesMigration = async (
+  event: DynamoDBStreamEvent,
+  userDataClient: UserDataClient,
+  businessesDataClient: BusinessesDataClient,
+  logger: LogWriterType
+): Promise<void> => {
+  for (const record of event.Records) {
+    logger.LogInfo(`Processing event ${record.eventName} with ID ${record.eventID}`);
+
+    if (
+      (record.eventName === "INSERT" || record.eventName === "MODIFY") &&
+      record.dynamodb &&
+      record.dynamodb.NewImage
+    ) {
+      const userId = record.dynamodb.NewImage.userId.S as string;
+      try {
+        const userData = await userDataClient.get(userId);
+
+        if (!userData || !userData.businesses || Object.keys(userData.businesses).length === 0) {
+          logger.LogError(`No businesses found for user ${userId}. Skipping.`);
+          continue;
+        }
+
+        const businessesData = userData.businesses;
+        for (const businessId in businessesData) {
+          const businessData = businessesData[businessId];
+          const existingBusiness = await businessesDataClient.get(businessId);
```

Comment:
Boulder - businessesDataClient no longer performs the migration function, we are taking out dated data and over writing the userData object.

Reply:
Within `migrateData` in `dynamoDBConfig.ts`, the line `const dataVersion = data.version ?? CURRENT_VERSION;` doesn't make sense to me. 

The methods used in `DynamoBusinessesDataClient` use this function. You previously mentioned the version field within the business object is used for tracking changed, but the methods in DynamoBusinessesDataClient` that use the fn pass in the business object.  We run into an issue b/c the version field in the business object is not the same as the version field for migrations.